ml-agents/mlagents/trainers/saver/tf_saver.py

```diff
@@ -0,0 +1,134 @@
+        # Save the TF checkpoint and graph definition
+        with self.graph.as_default():
+            if self.saver:
+                self.saver.save(self.sess, f"{checkpoint_path}.ckpt")
+            tf.train.write_graph(
+                self.graph, self.model_path, "raw_graph_def.pb", as_text=False
+            )
+        # also save the policy so we have optimized model files for each checkpoint
+        self.export(checkpoint_path, brain_name)
+        return checkpoint_path
+
+    def export(self, output_filepath: str, brain_name: str) -> None:
+        """
+        Saves the serialized model, given a path and brain name.
+
+        This method will save the policy graph to the given filepath.  The path
+        should be provided without an extension as multiple serialized model formats
+        may be generated as a result.
+
+        :param output_filepath: path (without suffix) for the model file(s)
+        :param brain_name: Brain name of brain to be trained.
+        """
+        export_policy_model(output_filepath, brain_name, self.graph, self.sess)
+
+    def maybe_load(self):
+        # If there is an initialize path, load from that. Else, load from the set model path.
+        # If load is set to True, don't reset steps to 0. Else, do. This allows a user to,
+        # e.g., resume from an initialize path.
+        reset_steps = not self.load
+        if self.initialize_path is not None:
+            self._load_graph(self.initialize_path, reset_global_steps=reset_steps)
+        elif self.load:
+            self._load_graph(self.model_path, reset_global_steps=reset_steps)
+        else:
+            self.policy._initialize_graph()
+
+    def _load_graph(self, model_path: str, reset_global_steps: bool = False) -> None:
+        with self.graph.as_default():
+            logger.info(f"Loading model from {model_path}.")
+            ckpt = tf.train.get_checkpoint_state(model_path)
+            if ckpt is None:
+                raise UnityPolicyException(
+                    "The model {} could not be loaded. Make "
+                    "sure you specified the right "
+                    "--run-id and that the previous run you are loading from had the same "
+                    "behavior names.".format(model_path)
+                )
+            try:
+                self.saver.restore(self.sess, ckpt.model_checkpoint_path)
+            except tf.errors.NotFoundError:
+                raise UnityPolicyException(
+                    "The model {} was found but could not be loaded. Make "
+                    "sure the model is from the same version of ML-Agents, has the same behavior parameters, "
+                    "and is using the same trainer configuration as the current run.".format(
+                        model_path
+                    )
+                )
+            self._check_model_version(__version__)
+            if reset_global_steps:
+                self.policy._set_step(0)
```

Comment:
_set_step is marked as protected but you are accessing it like a public method here. 
I think you should make set_step public